src/core/Carousel/CarouselDots.tsx

```diff
@@ -0,0 +1,138 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import { CarouselContext } from './CarouselContext';
+import { useTheme } from '../utils';
+import { CarouselDot } from './CarouselDot';
+import '@itwin/itwinui-css/css/carousel.css';
+
+type CarouselDotsProps = {
+  /** Number of total dots/slides in the carousel. Will be inferred from Carousel context. */
+  length?: number;
+  /** Maximum number of dots to show in the viewport. @default 5 */
+  maxCount?: number;
+  /** Index of currently acitve dot. Will be inferred from Carousel context. */
+  currentIndex?: number;
+  /** Callback fired when any of the dots are clicked. */
+  onSlideChange?: (index: number) => void;
+} & React.ComponentPropsWithoutRef<'div'>;
+
+/**
+ * The `CarouselDots` component shows a list of `CarouselDot` components which can be used to
+ * choose a specific slide. If used as a descendant of `Carousel`, then this component does not need
+ * any props or `children`.
+ *
+ * The props can be specified if this component is being used outside `Carousel`. `children` can be specified
+ * to override the individual dots that are shown.
+ *
+ * @example
+ * <Carousel>
+ *   // ...
+ *   <Carousel.Dots />
+ * </Carousel>
+ *
+ * @example
+ * <Carousel.Dots
+ *   length={10}
+ *   maxCount={3}
+ *   currentIndex={current}
+ *   onSlideChange={(i) => setCurrent(i)}
+ * />
+ */
+export const CarouselDots = React.forwardRef<HTMLDivElement, CarouselDotsProps>(
```

Comment:
Maybe rename to `CarouselDotsList` because for a few moments I thought that VS Code is showing me the same file in the list then I thought it might be some bug 😀

Reply:
Renamed